xarray/backends/netCDF4_.py

```diff
@@ -182,7 +182,10 @@ def _extract_nc4_variable_encoding(variable, raise_on_invalid=False,
 def _open_netcdf4_group(filename, mode, group=None, **kwargs):
     import netCDF4 as nc4
 
-    ds = nc4.Dataset(filename, mode=mode, **kwargs)
+    if isinstance(filename, nc4.Dataset):
+        ds = filename
```

Comment:
Not sure why that would be a problem. It's transmuting state, but IMO if you're handing xarray a `Dataset`, you kinda want it to do it's thing...